app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  before_action :set_comment, only: %i[show edit update destroy]
+
+  # GET /comments or /comments.json
+  def index
+    @post = Post.find(params[:post_id])
+    @comments = @post.comments.all
+    authorize @comments
+  end
+
+  # GET /comments/1 or /comments/1.json
+  def show
+    @post = Post.find(params[:post_id])
+    @comments = @post.comments.find(params[:id])
+  end
+
+  # GET /comments/new
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new
+    authorize @comment
+  end
+
+  # GET /comments/1/edit
+  def edit
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.find(params[:id])
+    authorize @comment
+  end
+
+  # POST /comments or /comments.json
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new(comment_params)
+
+    @comment.user_id = current_user.id
+    authorize @comment
+
+    respond_to do |format|
+      if @comment.save
+
+        format.turbo_stream
+        format.html { redirect_to post_path(params[:post_id]), notice: 'Comment was successfully created.' }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /comments/1 or /comments/1.json
+  def update
+    authorize @comment
+    respond_to do |format|
+      if @comment.update(comment_params)
+        format.html { redirect_to post_path(params[:post_id]), notice: 'Comment was successfully updated.' }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /comments/1 or /comments/1.json
+  def destroy
+    authorize @comment
+    respond_to do |format|
+      @comment.destroy!
```

Comment:
@aliraza4638 this is fine too but you can also use if & else just like above to follow consistency. Do read up `destroyed?`